### PR TITLE
Updating azure-cli extensions and extension version numbers

### PR DESCRIPTION
**Pull request by Gordon-BP**
## Description of changes

Azure recently released [an SSH extension for their CLI](https://learn.microsoft.com/en-us/cli/azure/ssh?view=azure-cli-latest) which did not have a nix package.

## Things done

* Ran the provided shell script to generate a description of the new ssh extension:

`bash query-extension-index.sh --download --nix --cli-version=2.26.0 --extension=ssh`

* Copy/pasted the output into `default.nix` in the right place with correct formatting

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/admin/azure-cli/extensions-generated.nix [hunk 1]
```diff
@@ -839,6 +839,13 @@
     sha256 = "ab19142c9a2e06190b6dac272d8cf29e179c9e283f965f8e3a4d29275b847ce5";
     description = "Microsoft Azure Command-Line Tools SiteRecovery Extension";
   };
+  ssh = mkAzExtension rec {
+    pname = "ssh";
+    version = "1.1.3";
```
Comment (techknowlogick) [inline pkgs/tools/admin/azure-cli/extensions-generated.nix:844]: Thanks for this PR! How did you choose `2.26.0` as the version to use in the `query-extension-index.sh` script? I think 2.60.0 is the latest CLI version included in nixpkgs. This is probably why `1.1.3` is the version for the ssh extension is chosen, and not 2.0.5: https://github.com/Azure/azure-cli-extensions/blob/main/src/ssh/setup.py#L10

**Comment by techknowlogick** [comment]
Also, you should target `master`, and then, based on the backport policy, it might be able to be backported to the 24.05 release channel (which is why you are getting a CI fail as it's checking the branch you are targetting).

One last note, for commits, it is generally recommended that you group related work into one single commit. and use a standardized commit message. In your case `azure-cli-extensions.ssh: init at x.y.z` would be the pattern for the commit message. Depending on your level of git comfort, you could squash your commits and amend the commit message, base your branch on `master`, and then force push.

File: pkgs/tools/admin/azure-cli/extensions-generated.nix (hunk 1 above)
Comment (Gordon-BP) [reply pkgs/tools/admin/azure-cli/extensions-generated.nix:844]: Hey good question! I think it is a typo. I checked the release notes page and the latest version is 2.63.0

https://learn.microsoft.com/en-us/cli/azure/release-notes-azure-cli

I'll go ahead and re-run the script with that version, thanks for the catch!

**Comment by Gordon-BP** [comment]
Closing this PR in favor of https://github.com/NixOS/nixpkgs/pull/334423 which targets `master`

**Pull request closed without merging**